### PR TITLE
Task#389698:[Rest][restprocessor] Generate "Permissions" section in yaml

### PR DESCRIPTION
**Pull request by v-haiboz**
Task#389698:[Rest][restprocessor] Generate "Permissions" section in yaml

### CODE REVIEW

File: Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs [hunk 1]
```diff
@@ -13,8 +14,8 @@
     public class RestOperationTransformer
     {
         private static ConcurrentDictionary<string, IList<Definition>> _cacheDefinitions = new ConcurrentDictionary<string, IList<Definition>>();
-
-        public static OperationEntity Transform(string groupKey, SwaggerModel swaggerModel, RestApiChildItemViewModel viewModel)
+        private static Regex _permissionRegex = new Regex("/providers(?<permission>(/[^/{}]+)+)(/{[^/}]+})*$", RegexOptions.IgnoreCase);
```
Comment (yishengjin1413) [inline Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs:17]: | RegexOptions.Compiled?

File: Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs [hunk 2]
```diff
@@ -51,6 +52,7 @@ public static OperationEntity Transform(string groupKey, SwaggerModel swaggerMod
 
             var basePath = swaggerModel.BasePath;
             var paths = TransformPaths(viewModel, scheme, host, basePath, apiVersion, allSimpleParameters);
+            var permission = TransformPermissions(viewModel, paths);
```
Comment (yishengjin1413) [inline Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs:55]: var permission = needPermission ?  TransformPermissions(viewModel, paths) : null;

File: Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs [hunk 3]
```diff
@@ -348,6 +351,46 @@ private static IList<PathEntity> TransformPaths(RestApiChildItemViewModel viewMo
             return pathEntities;
         }
 
+        private static string TransformPermissions(RestApiChildItemViewModel viewModel, IList<PathEntity> pathEntities)
+        {
+                var pathEntity = pathEntities.FirstOrDefault();
+                var index = pathEntity.Content.LastIndexOf("/providers/");
```
Comment (yishengjin1413) [inline Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs:357]: What if index == -1

Comment (v-haiboz) [reply Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs:357]: Don't do anything about it

for example
image

I want to get this string("/Microsoft.PolicyInsights/policyEvents/default/queryResults") and use it for regular verification

File: Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs (hunk 2 above)
Comment (v-haiboz) [reply Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs]: Updated

File: Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs (hunk 1 above)
Comment (v-haiboz) [reply Microsoft.RestApi.RestTransformer/RestOperationTransformer.cs]: Updated

**Review by yishengjin1413 — APPROVED** [review]

**Review by TianqiZhang — APPROVED** [review]

**Pull request merged**